src/Uri.php

```diff
@@ -103,6 +103,14 @@ public function __construct(string $uri = '')
      */
     private static function parse(string $url)
     {
+        // If IPv6
+        $prefix = '';
+        if (preg_match('%^(.*://\[[0-9:a-f]+\])(.*?)$%', $url, $matches)) {
+            /** @var array{0:string, 1:string, 2:string} */
```

Comment:
This shoud describe `$matches`. Currently it looks like you are describing `$prefix`